controllers/base_controller.js

```diff
@@ -151,8 +154,26 @@ module.exports = function BaseControllerModule(pb) {
             onlyThisSite: true,
             siteObj: this.siteObj
         };
+        if(pb.config.localization && pb.config.localization.db){
+            var opts = {
+                where: {_id: self.site}
+            };
+            var queryService = new pb.SiteQueryService({site: self.site, onlyThisSite: true});
+
+            queryService.q("localizations", opts, function (err, result) {
+                if (util.isError(err)) {
+                    pb.log.error(err);
+                }
+                if(result && result[0] && result[0].storage)
+                    pb.Localization.storage =  util.deepMerge(result[0].storage,pb.Localization.storage);
```

Comment:
This step is very redundant.  With every request you are updating keys in storage that were probably just updated by the previous request.  Did you test this to ensure that the merge successfully removed keys?  I'm not sure the merge will remove the deleted overrides if they had previously been merged before.